### ISSUE TITLE
Tasks

### ISSUE THREAD

**Issue by sven-luepke** [KnightTec/mlmi-project#18]
## 11.8.
- [x] Dowload full OASIS-3 sMRI data
- [x] Create dataset csv by matching alzheimer's diagnosis to MR session
- [x] Create bar chart with label count and modality counts
- [x] Create OASIS-3 2D dataset
---------------------------------------------
## 18.8.
- [x] Split into training & validation (90/10)
- [x] Run DenseNet model
- [x] Write function to extract dict for monai dataset (args: list of modalities)
- [x] CacheDataset for OASIS-3 AD
- [x] Create transformation pipeline (foreground crop, harmonize resolution & values)
- [x] Fix 2D dataset files (not fixed, just replaced)
- [x] Train 2D classifiers on alzheimers classification for each modality seperately using DenseNet
- [x] Write results to table
- [x] Multimodal transform pipeline (misalignment)
- [x] Decision level fusion using all classifiers (product & max fusion)
---------------------------------------------------------------
## 25.8.
- [x] Fix decision level fusiom transformation
- [x] Input level fusion Densnet Alzheimer's classifier
- [x] Dowload OASIS-3 MR sessions to cluster
- [x] Download IXI
- [x] OASIS-3 pre-processsing test subset
- [x] OASIS-3 pre-processing monai orientation + cropping + resolution
- [x] Download BraTS 2021 local + visualize
- [x] Registration test with dipy
- [x] Skull stripping test with FSL BET
## 1.9.
---------------------------------------------------------------
- [x] drop MR session with missing meta data (local)
- [x] Download MIRIAD to cluster
- [x] drop MR sessions with missing meta data (cluster)
- [x] Download BraTS 2021 task 1 dataset to cluster
- [x] Drop sessions without T1w
- [x] MONAI reorientation (drop scans when dim != 3) (local)
- [x] MONAI reorientation (drop scans when dim != 3) (cluster)
- [x] Download & visualize SRI-24 template
## 8.9.
---------------------------------------------------------------
- [x] resample test with dipy (is affine good?) -> yes
- [x] Run reorientation on cluster from terminal -> not faster :(
- [x] Visualize resampled MR session
- [x] resampling to T1w in monai pipeline
- [x] isotropic resampling in monai pipeline
- [x] CropForeground
- [x] Run full pipeline on cluster on oasis-3
- [x] Visualize resampling to SRI-24 template
- [x] Download + untar MIRIAD again
- [x] unpack brats on cluster
- [x] slides 8 sessions before and after MONAI pipeline
- [x] synthmorph registration to t1w (test)
- [x] skull stripping using synthstrip (test)
- [x] oasis-3 skull strip (test)
- [x] oasis-3 crop & pad (test)
- [x] oasis-3 skull strip (full)
- [x] oasis-3 crop & pad (full)
- [x] IXI group scans
- [x] MIRIAD monai pipeline
- [x] MIRIAD skull strip
- [x] MIRIAD crop & pad
- [x] Create table with dirty fusion experiments (all ablations + missing modality gauss)
- [x] cluster runtime facts slide
## 15.9.
---------------------------------------------------------------
- [x] get parameter count densenet
- [x] google fusion sheet stats function
- [x] Finish input fusion experiments old dataset (input + gauss)
- [x] Input fusion with old dataset and masking + loss curves (https://chat.openai.com/share/c55a24db-b3f4-406f-8df3-f9df2a08c464)
- [x] visualize SRI-24 template (3 axis, stripped/unstripped)
- [x] finish stripping oasis-3 cluster
- [x] synthmorph oasis-3 to T1w
- [x] visualize co-registration overlay before/after
- [x] oasis-3 crop & pad (180, 210, 155)
- [x] miriad crop & pad (180, 210, 155)
- [x] ufuk cluster access
- [x] visual verification 8 random subjects (oasis-3 & miriad) (slides)
## Final
----------------------------------------------------------------
- [x] Run unimodal training with with densenet hyperparams + tensorboard + normalizeIntensity
- [x] Normalize per channel instead of cross-channel
- [x] create clean oasis-3 AD dataset (local)
- [x] oasis-3 clean train-val split by subject avg label (local)
- [x] Rerun local with clean dataset (reduce learning rate?)
----------------------------------------------------------------
- [x] Create clean 2d oasis data on cluster
- [x] Create csv files for preprocessed data
----------------------------------------------------------------
- [x] Implement partial convolutions
- [x] Implement gating (feature attribution)
- [x] save best validation model
- [x] compute more classfication metrics (precision, recall, accuracy)
- [x] Run 5 input fusion methods on old data (all mods)
- [x] Run 5 input fusion methods (zeros, gauss, mask, pconv, gate) with all modalities on cluster with new data
----------------------------------------------------------------
## No time!
----------------------------------------------------------------
- [ ] oasis-3 pretrain sets (T1-T2, T1-T2*, T1-FLAIR, T1-MRA)
- [ ] Train AD-miriad classifier with 10%, 50% and 100% of training data (SwinViT) (weight_decay + LR from Swin -UNETR)
- [ ] Setup pre-training code (from BCTV monai)
----------------------------------------------------------------
- [ ] Multimodal contrastive pre-training a la SimCLR (T1, T2)
- [ ] Fine-tune AD classifier on swin-vit using pretrained simCLR model
- [ ] Pretrain T1-T2*
- [ ] Pretrain T1-FLAIR
- [ ] Pretrain T1-MRA
----------------------------------------------------------------
- [ ] Multimodal contrastive pre-training a la ImageBind (T1, T2)
- [ ] Fine-tune AD classifier on swin-vit using pretrained image bind model
----------------------------------------------------------------
# Guided Research Ideas
----------------------------------------------------------------
- [ ] fMRI integration
- [ ] Multimodal generative models --> generate MRA
- [ ] Modality impact
- [ ] Attention visualization per modality
- [ ] MLFlow
- [x] tensorboard
- [ ] synthmorph oasis-3 to SRI-24
- [ ] synthmorph MIRIAD to SRI-24
- [ ] Disentangled Representations
- [ ] Generative Models
- [ ] Representation Learning from Diffusion Models
- [ ] Meta Learning
- [ ] DINO input fusion destillation
----------------------------------------------------------------
- [ ] Wilcoxon test (see M3AE paper)
- [ ] Feature level fusion
- [ ] Input fusion with gating
- [ ] t-SNE plot of representations
- [ ] Repeat fusion experiments with clean 3D data
- [ ] Repeat pre-training experiment with 3D data
----------------------------------------------------------------
- [ ] Fill papers table
- [ ] Train Swin-UNETR with only T1w (10%, 20%, ...)
----------------------------------------------------------------
- [ ] Train Swin-UNETR with 10% of data (decision level fusion)
- [ ] Pre-Train ImageBind using SwinVit and OASIS-3 data --> fine-tune on BraTS (each modality seperately + decision level fusion)
- [ ] Pre-Train SimCLR using SwinVit (modality = augmentation) --> fine-tune on BraTS (each modality seperately + decision level fusion)
